src/test/java/StringManipulationTest.java

```diff
@@ -49,101 +55,137 @@ public void testRemoveNthCharacter1() {
     }
 
     @Test
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This message should have only the last character deleted");
+        assertEquals("This message should have only the last character delete", manipulatedstring.removeNthCharacter(56, false));
     }
 
     @Test
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This message should be unreadable");
+        assertEquals("Ths esag soud e nradbl", manipulatedstring.removeNthCharacter(3, false));
     }
 
     @Test
     public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This message is meant to be deleted");
+        assertEquals("", manipulatedstring.removeNthCharacter(1, false));
     }
 
     @Test
     public void testRemoveNthCharacter7() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("");
+        assertEquals("", manipulatedstring.removeNthCharacter(3, true));
     }
 
     @Test
     public void testGeSubStrings1() {
         manipulatedstring.setString("This is my string");
         String [] sStings = manipulatedstring.getSubStrings(3, 4);
 
-        assertEquals(sStings[0], "my");
-        assertEquals(sStings[1], "string");
+        assertEquals("my", sStings[0]);
+        assertEquals("string", sStings[1]);
     }
 
     @Test
     public void testGeSubStrings2() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This is another string");
+        String [] sStings = manipulatedstring.getSubStrings(1, 4);
+
+        assertEquals("This", sStings[0]);
+        assertEquals("string", sStings[3]);
     }
+
     @Test
     public void testGeSubStrings3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This is a odd string");
+        String [] sStings = manipulatedstring.getSubStrings(1, 3);
+
+        assertEquals("This", sStings[0]);
+        assertEquals("a", sStings[2]);
     }
+
     @Test
     public void testGeSubStrings4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This string is a very long string");
+        String [] sStings = manipulatedstring.getSubStrings(2, 4);
+
+        assertEquals("string", sStings[0]);
+        assertEquals("a", sStings[2]);
     }
+
     @Test
     public void testGeSubStrings5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This sting will return the first and last string");
+        String [] sStings = manipulatedstring.getSubStrings(1, 9);
+
+        assertEquals("This", sStings[0]);
+        assertEquals("string", sStings[8]);
     }
+
     @Test
     public void testGeSubStrings6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This string is the last test string");
```

Comment:
I'd recommend adding a test with a try/catch statement that checks that exceptions are thrown when necessary

Reply:
Yes it's a good idea there's 2 different exception in getSubString so I made test 2 throw a different exception as well